### PR TITLE
test: separate internal/external

### PR DESCRIPTION
**Pull request by Divlo**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[x] Other, please explain: Tests

**What changes did you make? (Give an overview)**

Separate the tests in 2 scripts : `test-internal` and `test-external`.

From @voxpelli :
> That way the badging situation will be easier to decipher – for maintainers as well as users and contributors.

:+1: 

**Which issue (if any) does this pull request address?**

Ref: https://github.com/standard/standard/issues/1663#issuecomment-870073653

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +68,9 @@
     "url": "git://github.com/standard/standard.git"
   },
   "scripts": {
-    "test": "./bin/cmd.js --verbose && tape test/*.js",
+    "test": "npm run test-internal && npm run test-external",
+    "test-internal": "./bin/cmd.js --verbose && tape test/internal/*.js",
```
Comment (voxpelli) [inline package.json:72]: Maybe we can keep the internal ones in the top folder? And have them be `test/*.js`? Less verbose?

**Review by voxpelli — APPROVED** [review]
LGTM 👍

File: package.json (hunk 1 above)
Comment (Divlo) [reply package.json]: Yes, it is less verbose! :+1: 
Done. :smile:

**Comment by voxpelli** [comment]
I'm 👍 on the changes

**Comment by voxpelli** [comment]
Random note: After this PR we should add `16.x` to the test matrix

**Review by duongxuantuan15 — CHANGES_REQUESTED** [review]
@Divlo

**Review by LinusU — APPROVED** [review]
Neat 👍

**Comment by Divlo** [comment]
> Random note: After this PR we should add `16.x` to the test matrix

Agreed. I already added `16.x` to the test matrix, is it fine or we should wait to do that after this PR ?

**Comment by voxpelli** [comment]
> > Random note: After this PR we should add `16.x` to the test matrix
> 
> Agreed. I already added `16.x` to the test matrix, is it fine or we should wait to do that after this PR ?

It’s fine I would say :)

**Review by juansonof01 — APPROVED** [review]

File: .github/workflows/test-external.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Test
+name: Test External
```
Comment (duongxuantuan15) [inline .github/workflows/test-external.yml:1]:
```suggestion
name: Test External
```

File: .github/workflows/test-external.yml [hunk 3]
```diff
@@ -7,12 +7,12 @@ on:
     branches: [master]
 
 jobs:
-  test:
+  test-external:
```
Comment (duongxuantuan15) [inline .github/workflows/test-external.yml:10]:
```suggestion
  test-external:
```

File: .github/workflows/test-external.yml [hunk 4]
```diff
@@ -7,12 +7,12 @@ on:
...
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        node-version: [10.x, 12.x, 14.x]
+        node-version: [10.x, 12.x, 14.x, 16.x]
```
Comment (duongxuantuan15) [inline .github/workflows/test-external.yml:15]:
```suggestion
        node-version: [10.x, 12.x, 14.x, 16.x]
```

File: .github/workflows/test-external.yml [hunk 5]
```diff
@@ -35,5 +35,5 @@ jobs:
       - name: Install dependencies
         run: npm install
 
-      - name: Run tests
-        run: npm test
+      - name: Run external tests
+        run: npm run test-external
```
Comment (duongxuantuan15) [inline .github/workflows/test-external.yml:39]:
```suggestion
        run: npm run test-external
```

File: .github/workflows/test-external.yml [hunk 6]
```diff
@@ -35,5 +35,5 @@ jobs:
       - name: Install dependencies
         run: npm install
 
-      - name: Run tests
-        run: npm test
+      - name: Run external tests
```
Comment (duongxuantuan15) [inline .github/workflows/test-external.yml:38]:
```suggestion
      - name: Run external tests
```

File: README.md [hunk 7]
```diff
@@ -8,7 +8,8 @@
 
 <p align="center">
   <a href="https://discord.gg/ZegqCBr"><img src="https://img.shields.io/discord/612704110008991783" alt="discord"></a>
-  <a href="https://github.com/standard/standard/actions?query=workflow%3ATest"><img src="https://github.com/standard/standard/workflows/Test/badge.svg" alt="status badge Node tests"></a>
+  <a href="https://github.com/standard/standard/actions/workflows/test-external.yml"><img src="https://github.com/standard/standard/workflows/test-external.yml/badge.svg?branch=master" alt="External tests"></a>
```
Comment (duongxuantuan15) [inline README.md:11]:
```suggestion
  <a href="https://github.com/standard/standard/actions/workflows/test-external.yml"><img src="https://github.com/standard/standard/workflows/test-external.yml/badge.svg?branch=master" alt="External tests"></a>
```

File: README.md [hunk 8]
```diff
@@ -8,7 +8,8 @@
...
+  <a href="https://github.com/standard/standard/actions/workflows/test-internal.yml"><img src="https://github.com/standard/standard/workflows/test-internal.yml/badge.svg?branch=master" alt="Internal tests"></a>
```
Comment (duongxuantuan15) [inline README.md:12]:
```suggestion
  <a href="https://github.com/standard/standard/actions/workflows/test-internal.yml"><img src="https://github.com/standard/standard/workflows/test-internal.yml/badge.svg?branch=master" alt="Internal tests"></a>
```

File: package.json [hunk 9]
```diff
@@ -68,7 +68,9 @@
     "url": "git://github.com/standard/standard.git"
   },
   "scripts": {
-    "test": "./bin/cmd.js --verbose && tape test/*.js",
+    "test": "npm run test-internal && npm run test-external",
```
Comment (duongxuantuan15) [inline package.json:71]:
```suggestion
    "test": "npm run test-internal && npm run test-external",
```

File: package.json [hunk 10]
```diff
@@ -68,7 +68,9 @@
...
+    "test-internal": "./bin/cmd.js --verbose && tape test/*.js",
```
Comment (duongxuantuan15) [inline package.json:72]:
```suggestion
    "test-internal": "./bin/cmd.js --verbose && tape test/*.js",
```

File: package.json [hunk 11]
```diff
@@ -68,7 +68,9 @@
...
+    "test-external": "tape test/external/*.js",
```
Comment (duongxuantuan15) [inline package.json:73]:
```suggestion
    "test-external": "tape test/external/*.js",
```

File: test/external/clone.js [hunk 12]
```diff
@@ -17,8 +17,8 @@ const testPkgs = require('./test.json')
 
 const GIT = 'git'
 const NPM = 'npm'
-const STANDARD = path.join(__dirname, '..', 'bin', 'cmd.js')
-const TMP = path.join(__dirname, '..', 'tmp')
+const STANDARD = path.join(__dirname, '..', '..', 'bin', 'cmd.js')
```
Comment (duongxuantuan15) [inline test/external/clone.js:20]:
```suggestion
const STANDARD = path.join(__dirname, '..', '..', 'bin', 'cmd.js')
```

File: test/external/clone.js [hunk 13]
```diff
@@ -17,8 +17,8 @@ const testPkgs = require('./test.json')
...
+const TMP = path.join(__dirname, '..', '..', 'tmp')
```
Comment (duongxuantuan15) [inline test/external/clone.js:21]:
```suggestion
const TMP = path.join(__dirname, '..', '..', 'tmp')
```

**Review by duongxuantuan15 — APPROVED** [review]
Ok

**Review by duongxuantuan15 — CHANGES_REQUESTED** [review]
@duongxuantuan15

**Review by duongxuantuan15 — APPROVED** [review]
@Divlo @voxpelli @LinusU @juansonof01 @juansonof01 @duongxuantuan15

**Comment by voxpelli** [comment]
@Divlo Something else you want to get fixed before merging?

**Comment by Divlo** [comment]
> @Divlo Something else you want to get fixed before merging?

Yes, `test external` with Node.js v16 is failing.
This is one of @feross repo that fails, see: https://github.com/feross/bitmidi.com/issues/116.

**Comment by voxpelli** [comment]
@Divlo lets remove `16.x` from this PR then, merge this PR, and then open a new one for that specific one :) This fix is too good to sit around here ;)

**Comment by Divlo** [comment]
> @Divlo lets remove `16.x` from this PR then, merge this PR, and then open a new one for that specific one :) This fix is too good to sit around here ;)

Let's do that! 😎
I opened a separate PR to add Node.js v16 : https://github.com/standard/standard/pull/1703

**Pull request merged**